### PR TITLE
feat: [PIE-4954]: Freeze Windows Init

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10834)

**Comment by ishantmehtaharness** [comment]
retrigger jest

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import NoResultsView from './views/NoResultsView/NoResultsView'
+// import { useUpdateQueryParams } from '@common/hooks'
+import css from './FreezeWindowsPage.module.scss'
+
+export default function FreezeWindowsPage(): React.ReactElement {
+  const { getString } = useStrings()
+  const { isGitSyncEnabled: isGitSyncEnabledForProject, gitSyncEnabledOnlyForFF } = useAppStore()
+  // const { updateQueryParams } = useUpdateQueryParams<{ templateType?: TemplateType }>()
+  const isGitSyncEnabled = isGitSyncEnabledForProject && !gitSyncEnabledOnlyForFF
+  // const history = useHistory()
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:36]: can we remove the commented code

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+  const { projectIdentifier, orgIdentifier, accountId } = useParams<ProjectPathProps>()
+  const [gitFilter, setGitFilter] = useState<GitFilterScope | null>(null)
+  const [, setPage] = useState(0)
+  const [searchParam, setSearchParam] = useState('')
+  const searchRef = React.useRef<ExpandingSearchInputHandle>({} as ExpandingSearchInputHandle)
+  const scope = getScopeFromDTO({ projectIdentifier, orgIdentifier, accountIdentifier: accountId })
+
+  useDocumentTitle([getString('common.freezeWindows')])
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:44]: This is not required when we do `pageName={PAGE_NAME.FreezeWindowsPage}` which we are already doing in the route definition

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+            items={[]}
+            placeholder={getString('all')}
+            popoverClassName={css.dropdownPopover}
+          />
+          {isGitSyncEnabled && (
+            <GitSyncStoreProvider>
+              <GitFilters
+                onChange={filter => {
+                  setGitFilter(filter)
+                  setPage(0)
```
Comment (udayvunnam) [inline src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:88]: is this to force rerender the page? Why is this required?

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+            }}
+            // value={templateType}
+            filterable={false}
+            addClearBtn={true}
+            // items={allowedTemplateTypes}
+            items={[]}
+            placeholder={getString('all')}
+            popoverClassName={css.dropdownPopover}
+          />
+          {isGitSyncEnabled && (
```
Comment (chetanvekariya-harness) [inline src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:83]: Check with Deepanshu on how to handle new gitsync. I think this handled only for old gitsync

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/views/NoResultsView/NoResultsView.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+export interface NoResultsViewProps {
+  hasSearchParam?: boolean
+  onReset?: () => void
+  text: string
+  minimal?: boolean
+  customImgSrc?: string
+}
+
+export default function NoResultsView({
```
Comment (chetanvekariya-harness) [inline src/modules/72-freeze-windows/pages/FreezeWindowsPage/views/NoResultsView/NoResultsView.tsx:26]: We have No Result related component. Could you check if that can be reused ? Some refactoring might be needed which we can do in next PR and use the same component.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx (hunk 1 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:36]: This is an ongoing feature, there might be some commented code. will be removed in subsequent PR's

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx (hunk 3 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:88]: Not force rerender. On changing a filter, we will land to page 0 and show those results

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx (hunk 4 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:83]: Sure, I will check with @depanshu-dhiman

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/views/NoResultsView/NoResultsView.tsx (hunk 5 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/views/NoResultsView/NoResultsView.tsx:26]: Sure chetan. Could you please point me to that component? I will reuse that in follow up PR

File: src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx (hunk 2 above)
Comment (ishantmehtaharness) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:44]: While using `pageName={PAGE_NAME.FreezeWindowsPage}`, Where do we define the document title for that page?

Comment (udayvunnam) [reply src/modules/72-freeze-windows/pages/FreezeWindowsPage/FreezeWindowsPage.tsx:44]: please ignore, its the `trackPage` that's automatically taken care by layout

**Review by udayvunnam — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]
